extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+
+
+def licenses(id):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if id is None:
+        return []
+    else:
+        return [{"license": {
+            "id": id
+        }}]
+
+
+def package_url(node: dict) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    qualifiers = {
+        "prev": node["prev"]
+    }
+    if node["user"]:
+        qualifiers["user"] = node["user"]
+    if node["channel"]:
+        qualifiers["channel"] = node["channel"]
+    if "#" in node["ref"]:
+        qualifiers["rref"] = node["ref"].split("#")[1]
```

Comment:
Note that the parsing of a reference is actually more complicated:
name/version@user/channel#rrev:pkgid#prev
https://docs.conan.io/2/reference/commands/list.html?highlight=prev
though I'm not sure whether they could appear in the graph json file.

Reply:
package id and prev are seperate feilds 

![image](https://github.com/conan-io/conan-extensions/assets/16867443/7e4d7755-5f2e-443a-9d6d-92a03974c93c)

Reply:
@prince-chrismc  Is the graph json format documented (other than just by this code https://github.com/conan-io/conan/blob/release/2.0/conans/client/graph/graph.py )?